src/helper_functions.cpp

```diff
@@ -25,14 +25,65 @@
   @date    25-May-18, D4N: created
  */
 
-#include "helper_functions.hpp"
+#include <config.h>
 
-#include <string.h>
+#include "helper_functions.hpp"
+#include "unused.h"
 
+#include <cassert>
+#include <cstring>
+#include <cwchar>
 
 std::string string_from_unterminated(const char* data, size_t data_length)
 {
     const size_t StringLength = strnlen(data, data_length);
 
     return std::string(data, StringLength);
 }
+
+std::string wideStr2Utf8(const std::wstring& wStr)
+{
+    std::mbstate_t state = std::mbstate_t();
+    const wchar_t* wstr = wStr.c_str();
+
+#ifdef EXV_HAVE_WCSRTOMBS_S
```

Comment:
According to this page:
https://en.cppreference.com/w/c/string/multibyte/wcsrtombs

`wcsrtombs_s` is available in c11. If we are requiring a c++11 compiler on master, we should not need that check. Or am I missing something?

Reply:
c11 != c++11. The C11 standard contains a lot of specs that are not included in C++11 (or even C++20), C++11 doesn't even include everything from C99 (especially the useful parts are missing...).

Reply:
Interesting ... I always assumed that features marked for `c11` were available in `c++11` and so far, I never had problems with the main compilers: gcc, clang, MSVC.

Reply:
Nope, then we could have already used the designated initializer in C++03:
``` C
struct foo {
  int a, b;
  size_t n;
};
static const struct foo foo_val = { .a = 1, .b = 2, .n = 5 };
```
Valid C99, invalid C++ unfortunately. This is the feature that I miss from C99 beside `restrict` (which every compiler needs for optimization anyway and thus adds `__restrict`) and flexible array members.